ksqldb-streams/src/test/java/io/confluent/ksql/execution/streams/SourceBuilderV1Test.java

```diff
@@ -0,0 +1,991 @@
+
+    // Then:
+    reset(buildContext);
+    stream.getExecutionKeyFactory().buildKeySerde(keyFormatInfo, PHYSICAL_SCHEMA, ctx);
+    verify(buildContext).buildKeySerde(keyFormatInfo, PHYSICAL_SCHEMA, ctx);
+  }
+
+  @Test
+  public void shouldReturnCorrectSerdeFactoryForWindowedSource() {
+    // Given:
+    givenWindowedSourceStream();
+
+    // When:
+    final KStreamHolder<?> stream = windowedStreamSource.build(planBuilder, planInfo);
+
+    // Then:
+    reset(buildContext);
+    stream.getExecutionKeyFactory().buildKeySerde(keyFormatInfo, PHYSICAL_SCHEMA, ctx);
+    verify(buildContext).buildKeySerde(keyFormatInfo, windowInfo, PHYSICAL_SCHEMA, ctx);
+  }
+
+  @Test
+  public void shouldBuildTableWithCorrectStoreName() {
+    // Given:
+    givenUnwindowedSourceTable(true);
+
+    // When:
+    tableSource.build(planBuilder, planInfo);
+
+    // Then:
+    verify(materializationFactory).create(keySerde, valueSerde, "base-Reduce");
+  }
+
+  @SuppressWarnings("unchecked")
+  private <K> ValueTransformerWithKey<K, GenericRow, GenericRow> getTransformerFromStreamSource(
+      final SourceStep<?> streamSource
+  ) {
+    streamSource.build(planBuilder, planInfo);
+    verify(kStream).transformValues(transformSupplierCaptor.capture());
+    final ValueTransformerWithKey<K, GenericRow, GenericRow> transformer =
+        (ValueTransformerWithKey<K, GenericRow, GenericRow>) transformSupplierCaptor.getValue().get();
+    transformer.init(processorCtx);
+    return transformer;
+  }
+
+  @SuppressWarnings("unchecked")
+  private <K> ValueTransformerWithKey<K, GenericRow, GenericRow> getTransformerFromTableSource(
+      final SourceStep<?> streamSource
+  ) {
+    streamSource.build(planBuilder, planInfo);
+    verify(kTable).transformValues(transformSupplierCaptor.capture());
+    final ValueTransformerWithKey<K, GenericRow, GenericRow> transformer =
+        (ValueTransformerWithKey<K, GenericRow, GenericRow>) transformSupplierCaptor.getValue().get();
+    transformer.init(processorCtx);
+    return transformer;
+  }
+
+  private void givenWindowedSourceStream(final int pseudoColumnVersion) {
+    when(buildContext.buildKeySerde(any(), any(), any(), any())).thenReturn(windowedKeySerde);
+    givenConsumed(consumedWindowed, windowedKeySerde);
+    windowedStreamSource = new WindowedStreamSource(
+        new ExecutionStepPropertiesV1(ctx),
+        TOPIC_NAME,
+        Formats.of(keyFormatInfo, valueFormatInfo, KEY_FEATURES, VALUE_FEATURES),
+        windowInfo,
+        TIMESTAMP_COLUMN,
+        SOURCE_SCHEMA,
+        OptionalInt.of(pseudoColumnVersion)
+    );
+  }
+  private void givenWindowedSourceStream() {
+    givenWindowedSourceStream(SystemColumns.CURRENT_PSEUDOCOLUMN_VERSION_NUMBER);
+  }
+
+  private void givenUnwindowedSourceStream(final int pseudoColumnVersion) {
+    when(buildContext.buildKeySerde(any(), any(), any())).thenReturn(keySerde);
+    givenConsumed(consumed, keySerde);
+    streamSource = new StreamSource(
```

Comment:
This test is not actually testing the V1 versions of any of these execution steps. It is currently identical to SourceBuilderTest.java.